sql/core/src/main/scala/org/apache/spark/sql/execution/Window.scala

```diff
@@ -920,7 +931,7 @@ private[execution] object AggregateProcessor {
     // Create the projections.
     val initialProjection = newMutableProjection(
       initialValues,
-      Seq(SizeBasedWindowFunction.n))()
+      Seq(partitionSize))()
```

Comment:
An initial value is allmost allways a constant. So in that case you don't need an input variable. `partitionSize` is a bit special because we know the value before the aggregates get initialized but not early enough to efficiently employ somekind of folding.